tests/functions/length.json

```diff
@@ -52,6 +52,18 @@
       "document" : [{"d": "f"}],
       "result": []
     },
+    {
+      "name": "filter on primitive array element, selects nothing",
+      "selector": "$[?length(@)==1]",
+      "document": [1],
+      "result": []
+    },
```

Comment:
Same as with https://github.com/jsonpath-standard/jsonpath-compliance-test-suite/pull/56#discussion_r1501902004; the intention was to make sure no library misinterprets `@` as representing the JSON array (instead of its elements).